library/bignum_mod.c

```diff
@@ -39,7 +39,7 @@ int mbedtls_mpi_mod_residue_setup( mbedtls_mpi_mod_residue *r,
                                    mbedtls_mpi_uint *p,
                                    size_t p_limbs )
 {
-    if( p_limbs < m->limbs || !mbedtls_mpi_core_lt_ct( m->p, p, p_limbs ) )
+    if( p_limbs != m->limbs || !mbedtls_mpi_core_lt_ct( p, m->p, m->limbs ) )
         return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
```

Comment:
note for later: since we require `p_limbs == m->limbs`, we could remove `p_limbs` from the input parameters to this function.

Reply:
I was thinking about that, but thought it is safer/more convenient, if the caller can pass the number of allocated limbs and this function always checks it automatically.

Reply:
I agree that in the use case you mention when you allocate and call this function immediately it is superfluous. For that use case we might add a convenience wrapper later that does the allocation as well.

There will be cases where we are using arrays in static memory and we calculate the allocated limbs by `sizeof()`. I think in this case it is better to double-check.

There might be cases at some point when we have a memory pool and we want to pass a chunk of that and the size of the chunk is calculated elsewhere, not where we are calling this. I am not sure whether we will do that or having this check is useful in this setting, just have the feeling that it might be.